### PR TITLE
txSyncer: service does not start after node catching up

### PR DESCRIPTION
**Pull request by algonautshant**
The txHandler and txSyncer are stopped when catching up from a catchpoint, and started again afterwards.

But they never really restart. The main loop of these services rely on a context to stop. That context is canceled when stopping. Then, when the services are started again, the contexts remain canceled, which leads for the service to immediately end.

The fix is to create a new context when starting these services.

txSync has another control mechanism to wait for before starting the loop, which should also be reset at (re)start.

This PR fixes the txSyncer part of #4775

### CODE REVIEW

File: catchup/service.go [hunk 1]
```diff
@@ -124,6 +124,7 @@ func MakeService(log logging.Logger, config config.Local, net network.GossipNode
 func (s *Service) Start() {
 	s.done = make(chan struct{})
 	s.ctx, s.cancel = context.WithCancel(context.Background())
+	atomic.SwapUint32(&s.initialSyncNotified, 0)
```
Comment (brianolson) [inline catchup/service.go:127]: 'swap' is the wrong operation here since you're throwing away the swapped-out value. `atomic.Store*()` ?

**Review by brianolson — COMMENTED** [review]
same pattern txHandler and txSyncer, eh?

**Comment by algonautshant** [comment]
The txHandler and txSync are fixed in separate PRs with separate tests to make sure each one is fixed and tested separately before putting them together. 
The fix for any of txHandler and txSync can mask the other one. 
The test for txHandler fix will fail with txSync alone, and the test for txSync will fail with txHandler fix alone.

**Review by brianolson — APPROVED** [review]

**Review by algorandskiy — APPROVED** [review]
Closes #4775

**Review by algorandskiy — APPROVED** [review]

**Pull request merged**